src/main/kotlin/com/launchdarkly/intellij/action/ToggleFlagAction.kt

```diff
@@ -51,31 +51,29 @@ class ToggleFlagAction : AnAction {
      * @param event Event received when the associated menu item is chosen.
      */
     override fun actionPerformed(event: AnActionEvent) {
-        val project = event.project
-        if (project != null) {
-            val selectedNode =
-                project.service<FlagToolWindow>().getPanel().tree.lastSelectedPathComponent as DefaultMutableTreeNode
-            val nodeInfo: FlagNodeParent = selectedNode.userObject as FlagNodeParent
-            // Relies on implicit behavior of key being first child.
-            val flagKey = selectedNode.firstChild.toString().substringAfter(" ")
-            val settings = LaunchDarklyMergedSettings.getInstance(project)
-            val patchComment = PatchComment()
-            val patch = PatchOperation()
-            patch.op = "replace"
-            patch.path = "/environments/" + settings.environment + "/on"
-            patch.value = !nodeInfo.env.on
-            patchComment.patch = listOf(patch)
-            val ldFlag = LaunchDarklyApiClient.flagInstance(project)
-            ApplicationManager.getApplication().executeOnPooledThread {
-                try {
-                    ldFlag.patchFeatureFlag(settings.project, flagKey, patchComment)
-                } catch (e: ApiException) {
-                    val notifier = GeneralNotifier()
+        val project = event.project ?: return
+        val selectedNode =
+            project.service<FlagToolWindow>().getPanel().tree.lastSelectedPathComponent as DefaultMutableTreeNode
+        val nodeInfo: FlagNodeParent = selectedNode.userObject as FlagNodeParent
+        // Relies on implicit behavior of key being first child.
+        val flagKey = selectedNode.firstChild.toString().substringAfter(" ")
+        val settings = LaunchDarklyMergedSettings.getInstance(project)
+        val patchComment = PatchComment()
+        val patch = PatchOperation()
```

Comment:
Now that we know of `apply` do we want to update this section to use that? Might make it cleaner to read.

Reply:
not sure i'm following... use apply how?